src/command_graph_generator.cc

```diff
@@ -581,7 +477,130 @@ void command_graph_generator::generate_distributed_commands(const task& tsk) {
 			}
 		}
 	}
+}
+
+void command_graph_generator::generate_distributed_commands(const task& tsk) {
+	const auto chunks = split_task_and_assign_chunks(tsk);
+	const auto chunks_with_requirements = compute_per_chunk_requirements(tsk, chunks);
+
+	resolve_pending_reductions(tsk, chunks_with_requirements);
+	generate_pushes(tsk, chunks_with_requirements);
+	generate_await_pushes(tsk, chunks_with_requirements);
+
+	// Union of all per-buffer writes on this node, used to determine which parts of a buffer are fresh/stale later on.
+	std::unordered_map<buffer_id, region<3>> per_buffer_local_writes;
+
+	// Collect all local chunks for detecting overlapping writes between all local chunks and the union of remote chunks in a distributed manner.
+	box_vector<3> local_chunks;
```

Comment:
Actually there was no point in collecting these at all, since we already have all local chunks in one place (in `chunks_with_requirements`) - removed.